tests/generative/test_query_model.py

```diff
@@ -112,14 +156,46 @@ def run_test(query_engines, data, variable, recorder):
         ), f"Mismatch between {first_name} and {second_name}"
 
 
+def run_error_test(query_engines, data, variable):
+    instances, variables = setup_test(data, variable)
+    for name, engine in query_engines.items():
+        try:
+            run_with(engine, instances, variables)
+        except Exception as e:  # pragma: no cover
+            assert False, f"{name} engine encountered an error: {e}"
```

Comment:
Sorry -- I don't understand the purpose of this. Doesn't it fail when we get expected errors.

Reply:
The point was to ensure all the errors are caught.  i.e. all the date operations in the `test_handle_date_errors` parameters generate out of range dates in some way. If they raise errors, `run_with` should catch them and return an `IGNORE_RESULT`.  It doesn't assert that, because some of the engines don't raise errors (e.g. sqlite returns None for operations that result in a year > 9999 although it errors for years of <= 0).  I had it catch the exception rather than leaving it to just raise it so that we can report the engine the error occurred with.

Reply:
It is in fact just that one - sqlite handles dates outside of the upper limit and returns None instead of an error, so I've revised the test to assert that the result is either `IGNORE_RESULT` or `{..., "v": None}` .  That would also have caught the error in the test for days (2000-01-01 + 8000*365 days was a valid calculation because of leap years - updated to + 8000*366 now)